File: LREAS/Server/real_time_quiz/models/Question.js
const mongoose = require('mongoose');

const QuestionSchema = new mongoose.Schema({
    quiz_version: { type: mongoose.Schema.Types.ObjectId, ref: 'QuizVersion', required: true },
    time_limit: { type: Number },
    date_updated: { type: Date, default: Date.now },
    score: { type: Number,default:100 },
    date_created: { type: Date, default: Date.now },
    position: { type: Number ,default:0},
    title: { type: String, required: true , default:"untitled"},
    image: { type: String }, // Image path
    answers: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Answer',default:[] }]
}, { timestamps: true });

module.exports = mongoose.model('Question', QuestionSchema,'question');

File: LREAS/Server/quiz/src/main/java/com/lreas/quiz/models/Question.java
package com.lreas.quiz.models;

import lombok.Getter;
import lombok.Setter;

import org.springframework.data.annotation.Id;

import org.springframework.data.annotation.ReadOnlyProperty;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DocumentReference;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.Date;
import java.util.List;

@Document
@Getter
public class Question {
    @Id
    private String id;

    @DocumentReference(lazy = true)
    @Field(name = "quiz_version")
    @Setter @Getter
    private QuizVersion quizVersion;

    @Field(name = "time_limit")
    @Setter @Getter
    private Double timeLimit;

    @Field(name = "date_updated")
    @Setter @Getter
    private Date dateUpdated;

    @Field(name = "score")
    @Setter @Getter
    private Double score;

    @Field(name = "date_created")
    @Setter @Getter
    private Date dateCreated;

    @Field(name = "position")
    @Setter @Getter
    private Integer position;

    @Field(name = "title")
    @Setter @Getter
    private String title;

    @Field(name = "image")
    @Setter @Getter
    private String image; // image path

    @ReadOnlyProperty
    @DocumentReference(lookup = "{'question': ?#{#self._id}}", lazy = true)
    @Setter @Getter
    private List<Answer> answers;
}
